blocks/product-gallery-block/features/product-gallery/_children/ProductFocusView/index.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState, useRef } from "react";
+
+import {
+	Button,
+	Icon,
+	Image,
+	Paragraph,
+	Stack,
+	imageANSToImageSrc,
+} from "@wpmedia/arc-themes-components";
+
+const BLOCK_CLASS_NAME = "b-product-gallery";
+
+const ProductFocusView = ({
+	initialItemId,
+	onClose,
+	productImages,
+	resizerAppVersion,
+	resizerURL,
+}) => {
+	const thumbnailBarItemLimit = 3;
+	const getImageIndexById = (id) => productImages.findIndex((image) => image._id === id);
+	const mainImagesRef = useRef([]);
+	const [selectedImageIndex, setSelectedImageIndex] = useState(getImageIndexById(initialItemId));
+	const [thumbnailBarStartIndex, setThumbnailBarStartIndex] = useState(selectedImageIndex);
+
+	const shouldShowDownButton = () =>
+		productImages.length >= thumbnailBarItemLimit &&
+		thumbnailBarStartIndex + thumbnailBarItemLimit < productImages.length;
+
+	const shouldShowUpButton = () => thumbnailBarStartIndex !== 0;
+
+	useEffect(() => {
+		let scrollTimeout;
+		const scrollImageIntoView = () => {
+			const currentImageContainer = mainImagesRef.current?.[selectedImageIndex];
+			if (currentImageContainer?.querySelector("img").complete) {
+				currentImageContainer.scrollIntoView();
+			} else {
+				scrollTimeout = setTimeout(scrollImageIntoView, 10);
+			}
+		};
+		scrollImageIntoView();
+		return () => clearTimeout(scrollTimeout);
+	}, [selectedImageIndex, mainImagesRef]);
+
+	return (
+		<div className={`${BLOCK_CLASS_NAME}__focus-view`}>
+			<Stack direction="horizontal" justifacation="start">
+				<div className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-bar`}>
+					<Paragraph>
+						{selectedImageIndex + 1} / {productImages.length}
+					</Paragraph>
+					{shouldShowUpButton() ? (
+						<Button
+							accessibilityLabel="Previous Image"
+							onClick={() => setThumbnailBarStartIndex(thumbnailBarStartIndex - 1)}
+						>
+							<Icon name="ChevronUp" />
+						</Button>
+					) : (
+						<div className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-bar-spacer`} />
+					)}
+					<Stack alignment="center" direction="vertical" justifacation="start">
+						{productImages
+							.filter(
+								(_, index) =>
+									index >=
+										Math.min(
+											thumbnailBarStartIndex,
+											productImages.length - thumbnailBarItemLimit
+										) && index < thumbnailBarStartIndex + thumbnailBarItemLimit
+								// as long as selected index is less than the length of the product images minus the window
+							)
+							.map((image) => (
+								<Image
+									// used as part of a page design so empty string alt text
+									alt=""
+									className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-image${
+										image._id === productImages[selectedImageIndex]._id ? "--selected" : ""
+									}`}
+									height={100}
+									key={image._id}
```

Comment:
`key`s should be unique.

Should just be a matter of appending the `_id`: `focus_view_thumbnail_${image._id}`

Reply:
@nschubach : Good call! Made this change.
I think the image `HY6LDPEW4BBFDLBYD4S3S7LZ3E` is coming from the original Product Gallery tests and that ID is being used multiple times, but I'm sure there's a way to make it unique.